storage/local/chunk.go

```diff
@@ -280,35 +276,48 @@ type chunkIterator interface {
 	// Whether a given timestamp is contained between first and last value
 	// in the chunk.
 	contains(model.Time) (bool, error)
-	// values returns a channel, from which all sample values in the chunk
-	// can be received in order. The channel is closed after the last
-	// one. It is generally not safe to mutate the chunk while the channel
-	// is still open. If a value is returned with error!=nil, no further
-	// values will be returned and the channel is closed.
-	values() <-chan struct {
-		model.SamplePair
-		error
-	}
+	// scan, value, and err implement a bufio.Scanner-like interface.  The
+	// scan method advances the iterator to the next value in the chunk and
+	// returns true if that worked. In that case, the value method will
+	// return the sample pair the iterator has advanced to. If the scan
+	// method returns false, either an error has occured or the end of the
+	// chunk has been reached. In the former case, the err method will
+	// return the error. In the latter case, the err method will return nil.
+	// Upon creation, the iterator is at position "minus one". After the
+	// first scan call, value will return the 1st value in the
+	// chunk. valueAtOrBeforeTime and rangeValues all modify the iterator
+	// position, too. They behave as if their return values were retrieved
+	// after a scan call, i.e. calling the value or err methods after a call
+	// to those methods will retrieve the same return value again (or the
+	// last value in the range in case of rangeValues), and subsequent scan
+	// calls will advance the iterator from there.
+	scan() bool
+	value() model.SamplePair
+	err() error
```

Comment:
This is very long and no one will read it or be able to focus that long. It's also not self-contained. I've to jump to `bufio.Scanner` and read there too for no good reason.

This is roughly what it should look like.
```go
// scan the next value in the chunk. Returns false if reading the next value failed
// or the end of the chunk was reached.
scan() bool
// value returns the last successfully scanned sample pair.
value() model.SamplePair
// err returns the last error that occurred on reading sample pairs from the chunk.
err() error
```

Generally, the side effect of `rangeValues` and `valueAtOrBeforeTime` should be documented in these methods too. I feel like having these non-trivial side effects across methods is an indicator that the interface design is sub-optimal.